pkg/webui/console/views/device-add/manual/form/validation-schema.js

```diff
@@ -0,0 +1,447 @@
+                  key: Yup.string()
+                    .length(16 * 2, Yup.passValues(sharedMessages.validateLength))
+                    .required(sharedMessages.validateRequired),
+                })
+              : Yup.object().strip(),
+          ),
+          nwk_s_enc_key: Yup.lazy(() =>
+            lwVersion >= 110
+              ? Yup.object().shape({
+                  key: Yup.string()
+                    .length(16 * 2, Yup.passValues(sharedMessages.validateLength))
+                    .required(sharedMessages.validateRequired),
+                })
+              : Yup.object().strip(),
+          ),
+        }),
+      })
+    },
+  ),
+})
+
+const macSettingsSchema = Yup.object({
+  mac_settings: Yup.object().when(
+    ['_activation_mode', 'supports_class_b', 'supports_class_c'],
+    (mode, isClassB, isClassC, schema) =>
+      schema.shape({
+        rx1_data_rate_offset: Yup.lazy(value => {
+          if (mode !== ACTIVATION_MODES.ABP || value === undefined) {
+            return Yup.number().strip()
+          }
+
+          return Yup.number()
+            .min(0, Yup.passValues(sharedMessages.validateNumberGte))
+            .max(7, Yup.passValues(sharedMessages.validateNumberLte))
+        }),
+        rx1_delay: Yup.lazy(delay => {
+          if (mode !== ACTIVATION_MODES.ABP || delay === undefined || delay === '') {
+            return Yup.number().strip()
+          }
+
+          return Yup.number()
+            .min(1, Yup.passValues(sharedMessages.validateNumberGte))
+            .max(15, Yup.passValues(sharedMessages.validateNumberLte))
+        }),
+        factory_preset_frequencies: Yup.lazy(frequencies => {
+          if (!Boolean(frequencies)) {
+            return Yup.array().strip()
+          }
+
+          return Yup.array()
+            .default([])
+            .test(
+              'is-valid-frequency',
+              sharedMessages.validateFreqNumberic,
+              factoryPresetFreqNumericTest,
+            )
+            .test(
+              'is-empty-frequency',
+              sharedMessages.validateFreqRequired,
+              factoryPresetFreqRequiredTest,
+            )
+        }),
+        rx2_frequency: Yup.lazy(frequency => {
+          if (frequency === undefined || frequency === '') {
+            return Yup.number().strip()
+          }
+          return Yup.number().min(100000, Yup.passValues(sharedMessages.validateNumberGte))
+        }),
+        rx2_data_rate_index: Yup.lazy(dataRate => {
+          if (dataRate === '' || dataRate === undefined) {
+            return Yup.number().strip()
+          }
+
+          return Yup.number()
+            .min(0, Yup.passValues(sharedMessages.validateNumberGte))
+            .max(15, Yup.passValues(sharedMessages.validateNumberLte))
+        }),
+        ping_slot_periodicity: Yup.lazy(value => {
+          if (isClassB) {
+            if (mode === ACTIVATION_MODES.MULTICAST || mode === ACTIVATION_MODES.ABP) {
```

Comment:
```suggestion
            if (mode === ACTIVATION_MODES.ABP) {
```

I might miss something but multicast should be irrelevant here since it will already imply that either Class B or Class C or both are set.

Reply:
That is to add the `required` requirement here for abp and multicast class B end devices.